### ISSUE TITLE
Octave has no GUI enabled (but has a desktop icon)

### ISSUE THREAD

**Issue by necessarily-equal** [NixOS/nixpkgs#123959]
**Describe the bug**
Octave's GUI is absent. Clicking on the desktop icon does nothing, and `octave --gui` errors out.

**To Reproduce**
I assume simply installing octave or running `nix-shell -p octaveFull --run "octave --gui"` is enough to reproduce. I can provide a `configuration.nix` if needed.

**Expected behavior**
I expect octave to start. Especially since it has a desktop icon.

**Additional context**
Maybe the `octave` (not `octaveFull`) package is not meant to have the GUI, but in that case it would probably be a good idea not to put the desktop icon in that package.

Note there are possibly two bugs here. Maybe "octave has a desktop icon without gui features", and/or "octaveFull can't start on GNOME".

**Notify maintainers**
@doronbehar @7c6f434c 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.109, NixOS, 20.09.3774.a6847cb5460 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-20.09.3774.a6847cb5460, nixpkgs-git"`
 - channels(afontain): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Maintainer information:**
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: [ octave octaveFull ]
```

**Comment by doronbehar** [comment]
I can't reproduce the issue with `octaveFull` - For me here on Gnome, both these commands launch the gui:

```
gio launch $(nix-build -A octaveFull)/share/applications/org.octave.Octave.desktop
nix-shell -p octaveFull --run "octave --gui"
```

Regarding the desktop icon issue, I think it does make sense for an Octave build that doesn't include gui support (the `octave` attribute), to have a desktop launcher. Though I agree that currently it does nothing and hence it doesn't make sense to have it. It should be changed a bit IMO:

```diff
diff --git i/org.octave.Octave.desktop w/org.octave.Octave.desktop
index dc3528a6528..275db3a82f9 100644
--- i/org.octave.Octave.desktop
+++ w/org.octave.Octave.desktop
@@ -11,9 +11,9 @@ Comment[nl]=Interactieve programmeeromgeving voor numerieke berekeningen
 Comment[pt]=Ambiente de programação interativo para computação numérica
 Comment[zh]=数值计算交互式编程环境
 TryExec=/nix/store/5ljp79b39ldhzngk4in7fwcqsqj4ndkd-octave-6.2.0/bin/octave
-Exec=/nix/store/5ljp79b39ldhzngk4in7fwcqsqj4ndkd-octave-6.2.0/bin/octave --gui %f
+Exec=/nix/store/5ljp79b39ldhzngk4in7fwcqsqj4ndkd-octave-6.2.0/bin/octave %f
 Icon=octave
-Terminal=false
+Terminal=true
 Type=Application
 Categories=Education;Science;Math;
 StartupNotify=false
```

I'm considering this is something worth suggesting upstream.

**Comment by necessarily-equal** [comment]
Ok, I've updated to 21.05 and the octaveFull derivation now works. Or have all the needed dependencies, not sure.

About the desktop icon, I've thought about it a bit, and I agree it would make more sense for upstream to build a desktop icon with the above change if it's missing the `--with-qt=5` configure flag.

**Comment by doronbehar** [comment]
> About the desktop icon, I've thought about it a bit, and I agree it would make more sense for upstream to build a desktop icon with the above change if it's missing the `--with-qt=5` configure flag.

I agree, I edited the initial comment of the issue and the title to reflect that.
